### PR TITLE
feat:[#2756, #2757] Add new categorisation and horizontal

### LINKED ISSUE #2756: Add the new categorisation to the EOSC Marketplace
The new categorization was applied by PC and is available in PC beta. Add the new categorization to the EOSC MP.

### PR DESCRIPTION
**Pull request by goreck888**
Add new fields for horizontal services
and the new research categorisation
Closes #2756, #2757

NOTICE!
New fields are available in the service form under the "classification" section

### CODE REVIEW

**Comment by Marcelinna** [comment]
Steps to reproduce 
1. Select Horizontal Service checkbox
2. Click on the button "Update Resource"
3. Click on the button "Edit Resource"
4. Unselect Horizontal Service checkbox
5. Click on the button "Update Resource"
6. See problem - checkbox is selected

**Comment by Marcelinna** [comment]
changelog

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce

1. Set horizontal service to true for one of services
2. go to /services page
3. See a problem 
image

**Comment by Marcelinna** [comment]
Step to reproduce

1. Update provider in PC - provider is visible in MP
2. Update provider in PC - provider is duplicated in MP 
image

**Comment by Marcelinna** [comment]
changelog

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
 After import resource has upstream set to MP, upstream should be set to eosc-registry
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

The approved resource is not visible in MP

1. Create resources in PC
2. Approve resource in PC
3. See problem - resource is not visible in MP

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

In MP I have provider **provider_0310_1** and his copy **eosc.provider_0310_1**
1. Import resource eosc.provider_0310_1.provider_0310_1_2
2. See problem - resource is only visible for  **provider_0310_1**

**Comment by Marcelinna** [comment]
It ok that provider url is with eosc

https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/providers/**eosc.provider_1010**

resource url is without eosc

https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/services/r**esource-provider-10-10-update**

better if there was one approach, e.g. without eosc?

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce
1. Create a service in PC
1. Add a research category for created service
2. Go to Backoffice 
3. See problem, research category for service is empty

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Add horizontal service label to service page

image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce
1. Select provider with one resource
2. See problem - despite the provider has one resource, on services list are visible two services - one horizontal

image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce
1. Go to services
2. Set sort by A-Z
3. See problem - Horizontal services are between services ranked in order A-Z
image


Horizontal services should be added between services when sorting by Best result is set

**Comment by roksanaer** [comment]
@Marcelinna Gorek is right for the horizontal services in a->z sorting

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce
1. Create new provider in PC
2. Approve Provider in PC
3. Go to MP - provider page is crashed
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Step to reproduce
1. Mark service as horizontal in PC - in MP service has horizontal label
2. Unmark service as horizontal - see problem in MP resource still has horizontal label

the same situation for research category

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Now horizontal service occurs every 9, did we not agree with every 5 ?

File: lib/tasks/service_extras.rake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        service = Service.find_by(name: entry["name"])
+        if service.present?
+          service.update!(horizontal: entry["horizontal"], research_categories: mapped_categories)
+        else
+          puts "WARN: #{entry["name"]} not updated, service is blank: #{service}"
+        end
+      end
+    end
+  end
+end
```
Comment (Marcelinna) [inline lib/tasks/service_extras.rake:24]: this does not seem necessary. This would be useful if we were downloading data from a csv file.?

Comment (goreck888) [reply lib/tasks/service_extras.rake:24]: Yes, will be deleted soon

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Step to reproduce
1. Deactivate service in PC
2. See problem - in MP service is published

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
1. Create a resource with a flag horizontal in BO
2. See problem - resource is draft but it is visible in MP

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Step to reproduce
1. Create Provider in PC
2. Approve Provider in PC
3. Create first resource in PC
4. Approve first resource in PC
5. See problem - first resource has the status active, but not visible in MP

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
- Step to reproduce 
1. Go to the resource page
2. See problem - despite Items on page is set to 10, there are only 8 resources listed on first page
image

**Comment by Marcelinna** [comment]
I MP I have services 4 services with horizontal flag
1. Horizontal 1
2. Horizontal 2
3. Horizontal 3
4. Horizontal 4

When I go to the /services page and has set Items on page set to 10, on the result list 

on first page I see beetwen services: Horizontal 4, Horizontal 2
on second page I see: Horizontal 1, Horizontal 2
on thrid page I see: Horizontal 4, Horizontal 2
on the fourth page I see: Horizontal 2, Horizontal 3


@agpul should the same horizontal services show up on different pages? should they appear on every page and repeat ?
or if I have 4 horizontal services in the system, on the result list I should see only 4 horizontal services?

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce
1. Go to Backoffice
2. Select 3 Services as Horizontal
3. Go to MP
4. See problem - why horizontal services selected in the Backoffice are displayed one after the other, different behaviour from those created in the PC
image

**Review by michal-szostak — APPROVED** [review]

**Review by wziajka — APPROVED** [review]

**Pull request merged**